### DISCUSSION TITLE
Controller 嵌入 NameServer 部署方式，只有两个节点，停掉broker主节点，从节点不能变成主？

### DISCUSSION THREAD

**Discussion by onejimmyboy** [apache/rocketmq#7136] [Q&A] [answered]
版本：5.1.3
集群形态：Controller嵌入Namesrv部署，一共两个节点，即：controller2节点（嵌入namesrv），broker2节点（1主一从）
现象：启动namesrv、broker。当前集群正常，一主一从。如果stop主节点，此时从节点无法变成主节点。这种情况是否正常。报错信息如下
![image](https://github.com/apache/rocketmq/assets/57432151/10ac5ce3-1755-499b-a730-8105464f402c)
节点1的broker和namesrv的配置
![image](https://github.com/apache/rocketmq/assets/57432151/a2b58143-421d-4d2a-8462-bf508f9fe4fb)
节点2的broker和namesrv的配置
![image](https://github.com/apache/rocketmq/assets/57432151/07155abe-8743-4699-b4d5-5c1eab9700d9)
我的疑问点：这种现象正常吗？

**Comment by RongtongJin** [comment]
@onejimmyboy 看配置没有问题（如果希望不丢消息，建议allAckInSyncStateSet设置为true），可以用getSyncStateSet命令看下，主备是否同步，怀疑切换时备并不是完全同步主，导致没有正常切换。

**Reply by onejimmyboy** [reply]
集群正常时执行命令行
![image](https://github.com/apache/rocketmq/assets/57432151/e2ca75fc-a4f7-4fec-bec5-caf079817ad7)
集群正常的时候，应该就没有同步，是不是两节点的这种情形不支持啊
把allAckInSyncStateSet设置为了true结果一样

**Reply by RongtongJin** [reply]
看下store.log，查下为啥没有正常同步。两节点是支持的

**Reply by onejimmyboy** [reply]
从节点中日志信息
![image](https://github.com/apache/rocketmq/assets/57432151/b7e43934-b17a-4e9a-b90d-c9228d21efef)
主节点日志信息
![image](https://github.com/apache/rocketmq/assets/57432151/39c95e8b-8831-4aa8-a3f0-39df5fdd88e8)

**Reply by RongtongJin** [reply]
日志显示从节点无法连接到主节点，检查一下连通性

**Reply by onejimmyboy** [reply]
我们主机用的是双网卡，在配置broker和namesrv的配置文件的时候，我使用的都是内网，192.168.x.x，但是不知道为啥store.log里面的日志确显示的是另外一个网段100.100.x.x。按理说，我配置的是192.168.x.x，store.log中的日志连接使用的也应该是192.168.x.x。而不是100.100.x.x。我排查了一遍配置文件，配置ip的地方都没有写100.100.x.x。（100.100.x.x主机间不互通，只有内网192.168.x.x主机间互通）

**Reply by onejimmyboy** [reply]
![image](https://github.com/apache/rocketmq/assets/57432151/1c9d5653-4383-461e-a132-960e479a8c59)
![image](https://github.com/apache/rocketmq/assets/57432151/a7bb93e0-d36b-4762-8d78-288e26b5345c)

这个是详细日志，
从上面的日志中可以看出来，最开始获取的masterAddress地址是正确的，但是再获取masterHaAddress确是100.100.6.78？

**Reply by onejimmyboy** [reply, marked as answer]
破案了，broker配置文件中没有配置brokerIP2,添加上brokerIP2后，正常切换了，brokerIP2=brokerIP1
